html5/js/OffscreenDecodeWorker.js

```diff
@@ -257,147 +78,66 @@ class WindowDecoder {
     send_decode_error(packet, message);
   }
 
-  packet_decoded(packet) {
-    try {
-      const coding = packet[6];
-      const packet_sequence = packet[8];
-      const start = this.pending_decode.get(packet_sequence);
-      if (!this.pending_decode.delete(packet_sequence)) {
-        //already cancelled somehow
-        return;
-      }
-      if (coding == "throttle") {
-        // Encoding throttle is used to slow down frame input
-        const timeout = 500;
-        setTimeout(() => {
-          this.send_decode_ok(packet, start);
-        }, timeout);
-      } else {
-        this.send_decode_ok(packet, start);
-      }
-      if (this.closed) {
-        return;
-      }
-
-      this.paint_packet(packet);
-
-      //any snapshot is now out of date:
-      this.snapshot_buffer = null;
-
-      //are there any packets still waiting to be decoded?
-      if (this.flush_seqs.length === 0) {
-        return;
-      }
-      //for the current flush sequence?
-      const flush_seq = this.flush_seqs[0];
-      const decode_wait = [...this.pending_decode.keys()].filter(
-        (x) => x <= flush_seq
-      );
-      if (decode_wait.length > 0) {
-        //yes, found some pending decodes with a sequence number lower than the current flush
-        return;
-      }
-      //we're going to have a snapshot ready (when using the back buffer),
-      //or we're going to schedule a take_snapshot()
-      //so we can safely cancel the current timer:
-      this.cancel_snapshot_timer();
-
-      this.schedule_show_frame();
-    } catch (error) {
-      console.error("error handling decoded packet:", error);
-      this.decode_error(packet, error);
+  queue_draw_packet(packet) {
+    if (this.closed) {
+      return;
+    }
+    this.decode_queue.push(packet);
+    if (!this.decode_queue_draining) {
+      this.proccess_decode_queue();
     }
   }
 
-  schedule_show_frame() {
-    //move to the next frame at the next vsync:
-    this.animation_request = vsync
-      ? requestAnimationFrame((t) => {
-          this.animation_request = 0;
-          this.next_frame();
-        })
-      : setTimeout(() => {
-          this.animation_request = 0;
-          this.next_frame();
-        }, 16);
```

Comment:
So now we make no effort to sync updates with the screen refresh?
This is not a deal breaker, but it would be nice to be able to try to sync up - somehow, I have no idea how!

Reply:
requestAnimationFrame is now supported in workers (and implemented in the code)

Reply:
Ah, I must have missed it.